sql_test.go

```diff
@@ -150,7 +151,8 @@ func TestRegisterDBStatsMetrics(t *testing.T) {
 	assert.NoError(t, err)
 
 	// Should collect 7 metrics
-	got, err := r.Collect(context.Background())
+	got := &metricdata.ResourceMetrics{}
+	err = r.Collect(context.Background(), got)
```

Comment:
I thought the original style was quite clear. Any reason to change this?

Reply:
API changes. See 
https://github.com/open-telemetry/opentelemetry-go/pull/3732